extension/httpfs/httpfs.cpp

```diff
@@ -499,8 +500,25 @@ void HTTPFileHandle::Initialize(FileOpener *opener) {
 			length = 0;
 			return;
 		} else {
-			throw IOException("Unable to connect to URL \"" + path + "\": " + to_string(res->code) + " (" + res->error +
-			                  ")");
+			// HEAD request fail, use Range requet for another try (read only one byte)
+			if ((flags & FileFlags::FILE_FLAGS_READ)) {
+				char dummy_buffer[2];
+				auto range_res = hfs.GetRangeRequest(*this, path, {}, 0, dummy_buffer, 2);
+				if (range_res->code != 206) {
+					throw IOException("Unable to connect to URL \"" + path + "\": " + to_string(res->code) + " (" +
+					                  res->error + ")");
+				}
+				auto range_find = range_res->headers["Content-Range"].find("/");
+				range_length = range_res->headers["Content-Range"].substr(range_find + 1);
```

Comment:
Can we have a check/assertion that `range_find != std::string::npos`, because `substr` will throw an exception if the first argument is > size

Reply:
I think this is a valid point, throwing a descriptive IOException when the Content-Range header is malformed seems like a good idea.